web/beacon-app/src/features/onboarding/components/steps/workspace/form.tsx

```diff
@@ -35,12 +35,12 @@ const WorkspaceForm = ({
     };
   }, [touched.workspace, setFieldValue, values, touched]);
 
-  // set error if workspace is already taken
+  // Set the error if backend returns a validation error.
   useEffect(() => {
     if (hasError) {
       setFieldError(
         'workspace',
-        t`The workspace URL is taken by another team. Try a variation or another slug.
+        t`The workspace name must be at least 3 characters and cannot start with a number.
```

Comment:
This error message should be displayed when a 409 status code is encountered. In that case, shouldn't this message be displayed? `The workspace URL is taken by another team. Try a variation or another slug`

Reply:
you could alternatively make these changes too by adding one new props for the 400 error.
 
 ```jsx
  type WorkspaceFormProps = {
  onSubmit: (values: any, helpers: FormikHelpers<any>) => void;
  isDisabled?: boolean;
  isSubmitting?: boolean;
  initialValues?: any;
  hasError?: boolean;
  hasValidationError:  boolean  /// in the parent component check the 400 error 
  shouldDisableInput?: boolean;
};
```